litellm/proxy/db/db_transaction_queue/spend_log_cleanup.py

```diff
@@ -56,35 +103,40 @@ async def cleanup_old_spend_logs(self, prisma_client: PrismaClient) -> None:
                 verbose_proxy_logger.error("Retention seconds is None, cannot proceed with cleanup")
                 return
 
-            cutoff_date = datetime.now(UTC) - timedelta(seconds=float(self.retention_seconds))
-            verbose_proxy_logger.info(f"Deleting logs older than {cutoff_date.isoformat()}")
-
-            total_deleted = 0
-            run_count = 0
-            while True:
-                if run_count > 100:
-                    verbose_proxy_logger.info("Max logs deleted - 1,00,000, rest of the logs will be deleted in next run")
-                    break
-                # Step 1: Find logs to delete
-                logs_to_delete = await prisma_client.db.litellm_spendlogs.find_many(
-                    where={"startTime": {"lt": cutoff_date}},
-                    take=self.batch_size,
+            # If we have a pod lock manager, try to acquire the lock
+            if self.pod_lock_manager and self.pod_lock_manager.redis_cache:
+                lock_acquired = await self.pod_lock_manager.acquire_lock(
+                    cronjob_id=SPEND_LOG_CLEANUP_JOB_NAME,
                 )
-                verbose_proxy_logger.info(f"🗑️ Found {len(logs_to_delete)} logs in this batch")
+                verbose_proxy_logger.info(f"Lock acquisition attempt: {'successful' if lock_acquired else 'failed'}  at {datetime.now()}")
 
-                if not logs_to_delete:
-                    verbose_proxy_logger.info(f"No more logs to delete. Total deleted: {total_deleted}")
-                    break
+                if not lock_acquired:
+                    verbose_proxy_logger.info("Another pod is already running cleanup")
+                    return
 
-                request_ids = [log.request_id for log in logs_to_delete]
+            try:
+                cutoff_date = datetime.now(timezone.utc) - timedelta(seconds=float(self.retention_seconds))
+                verbose_proxy_logger.info(f"Deleting logs older than {cutoff_date.isoformat()}")
 
-                # Step 2: Delete them in one go
-                await prisma_client.db.litellm_spendlogs.delete_many(
-                    where={"request_id": {"in": request_ids}}
-                )
+                # Perform the actual deletion
+                total_deleted = await self._delete_old_logs(prisma_client, cutoff_date)
+
+                verbose_proxy_logger.info(f"Deleted {total_deleted} logs")
+
+                # If we have a pod lock manager, release the lock
+                if self.pod_lock_manager and self.pod_lock_manager.redis_cache:
+                    await self.pod_lock_manager.release_lock(cronjob_id=SPEND_LOG_CLEANUP_JOB_NAME)
+                    verbose_proxy_logger.info(f"Released cleanup lock {datetime.now()}")
+                return  # Explicitly return after cleanup is complete
+
+            except Exception as e:
+                verbose_proxy_logger.error(f"Error during cleanup: {str(e)}")
```

Comment:
you can release the lock in a finally section, that way there is no dup code in try/except blocks